### PR TITLE
feat: delete version testing

### PR DESCRIPTION
**Pull request by zekth**
Following https://github.com/fastify/fastify/pull/2549 , delete the version check as now it's in Fastify itself.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: plugin.js [hunk 1]
```diff
@@ -1,47 +1,51 @@
...
 
-  if (typeof fn.default !== 'undefined') { // Support for 'export default' behaviour in transpiled ECMAScript module
+  if (typeof fn.default !== 'undefined') {
+    // Support for 'export default' behaviour in transpiled ECMAScript module
     fn = fn.default
   }
 
   if (typeof fn !== 'function') {
-    throw new TypeError(`fastify-plugin expects a function, instead got a '${typeof fn}'`)
+    throw new TypeError(
```
Comment (fox1t) [inline plugin.js:16]: Do you use different formatting rules?

**Review by fox1t — CHANGES_REQUESTED** [review]
LGTM

File: plugin.js (hunk 1 above)
Comment (zekth) [reply plugin.js:16]: Standard was saying there was errors, just applied `standard --fix`

**Comment by mcollina** [comment]
Is this semver-major or not? I might be tempted to ship it as minor.

File: plugin.js (hunk 1 above)
Comment (fox1t) [reply plugin.js:16]: Hmm, fair enough XD

**Comment by L2jLiga** [comment]
It breaks plugin validation unless newer version of fastify is installed so it's major IMO

**Comment by zekth** [comment]
> It breaks plugin validation unless newer version of fastify is installed so it's major IMO

That's my concern yes as long as there is no pointer to Fastify instance to check if it's `>=3.4`

File: package.json [hunk 2]
```diff
@@ -25,14 +25,12 @@
   "homepage": "https://github.com/fastify/fastify-plugin#readme",
   "devDependencies": {
     "@types/node": "^13.13.2",
-    "fastify": "^3.3.0",
+    "fastify": "^3.4.0",
```
Comment (jsumners) [inline package.json:28]: Let the release team adjust this after the PR has been solidified.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Review by fox1t — APPROVED** [review]

**Comment by mcollina** [comment]
@jsumners @delvedor any options on this? I would like to ship this asap.

**Review by delvedor — APPROVED** [review]
LGTM

**Review by jsumners — APPROVED** [review]
LGTM

**Pull request merged**